### DISCUSSION TITLE
wget not working on Sonoma

### DISCUSSION THREAD

**Discussion by Upperholme** [Homebrew/discussions#4819] [Getting started]
### Output of `brew config`

```shell
HOMEBREW_VERSION: 4.1.13
ORIGIN: https://github.com/Homebrew/brew
HEAD: a8519f78fb63f2f2266950bdd8141037da69f8bd
Last commit: 5 days ago
Core tap origin: https://github.com/Homebrew/homebrew-core
Core tap HEAD: e995843b010e1048b7b021117abe6a449a4c4614
Core tap last commit: 30 minutes ago
Core tap branch: master
Core tap JSON: 30 Sep 16:39 UTC
HOMEBREW_PREFIX: /usr/local
HOMEBREW_CASK_OPTS: []
HOMEBREW_MAKE_JOBS: 8
Homebrew Ruby: 2.6.10 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: octa-core 64-bit icelake
Clang: 12.0.0 build 1200
Git: 2.24.3 => /Library/Developer/CommandLineTools/usr/bin/git
Curl: 8.1.2 => /usr/bin/curl
macOS: 14.0-x86_64
CLT: 12.0.0.26.2
Xcode: N/A
```

### Output of `brew doctor`

```shell
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: Your Command Line Tools are too outdated.
Update them from Software Update in System Settings.

If that doesn't show you any updates, run:
  sudo rm -rf /Library/Developer/CommandLineTools
  sudo xcode-select --install

Alternatively, manually download them from:
  https://developer.apple.com/download/all/.
You should download the Command Line Tools for Xcode 15.0.


Warning: A newer Command Line Tools release is available.
Update them from Software Update in System Settings.

If that doesn't show you any updates, run:
  sudo rm -rf /Library/Developer/CommandLineTools
  sudo xcode-select --install

Alternatively, manually download them from:
  https://developer.apple.com/download/all/.
You should download the Command Line Tools for Xcode 15.0.


Warning: You have unlinked kegs in your Cellar.
Leaving kegs unlinked can lead to build-trouble and cause formulae that depend on
those kegs to fail to run properly once built. Run `brew link` on these:
  tcl-tk

Warning: Homebrew's "sbin" was not found in your PATH but you have installed
formulae that put executables in /usr/local/sbin.
Consider setting your PATH for example like so:
  echo 'export PATH="/usr/local/sbin:$PATH"' >> ~/.zshrc
```

### Description of issue

When I run wget I get:

dyld[42297]: Library not loaded: /usr/local/opt/libunistring/lib/libunistring.2.dylib
  Referenced from: <94F7987D-8C50-34E7-B456-452D70425F9F> /usr/local/Cellar/libidn2/2.3.4/lib/libidn2.0.dylib
  Reason: tried: '/usr/local/opt/libunistring/lib/libunistring.2.dylib' (no such file), '/System/Volumes/Preboot/Cryptexes/OS/usr/local/opt/libunistring/lib/libunistring.2.dylib' (no such file), '/usr/local/opt/libunistring/lib/libunistring.2.dylib' (no such file), '/usr/local/lib/libunistring.2.dylib' (no such file), '/usr/lib/libunistring.2.dylib' (no such file, not in dyld cache), '/usr/local/Cellar/libunistring/1.1/lib/libunistring.2.dylib' (no such file), '/System/Volumes/Preboot/Cryptexes/OS/usr/local/Cellar/libunistring/1.1/lib/libunistring.2.dylib' (no such file), '/usr/local/Cellar/libunistring/1.1/lib/libunistring.2.dylib' (no such file), '/usr/local/lib/libunistring.2.dylib' (no such file), '/usr/lib/libunistring.2.dylib' (no such file, not in dyld cache)
zsh: abort      wget

**Comment by SMillerDev** [comment]
Try `brew install libunistring`

**Comment by Upperholme** [comment]
Warning: libunistring 1.1 is already installed and up-to-date.

**Comment by Upperholme** [comment]
It looks like libidn.2.0.dylib is looking for libunistring.2.dylib, but my libunistring directory, at /usr/local/Cellar/libunistring/1.1/lib/ contains a file called  libunistring.5.dylib

(thanks for responding to my query, btw)

**Reply by ZhongRuoyu** [reply]
Your `libidn` is likely outdated. Does `brew upgrade libidn` get you a newer version?

If that fixes your issue, I recommend running `brew upgrade` to make sure you don't get mismatched dependencies. `libidn` should have been automatically updated when you updated `libidn` either manually or automatically as a dependency of something else.

If that doesn't fix it, what's the output of `brew linkage wget`?

**Comment by Upperholme** [comment]
Thanks.
So I ran

```brew upgrade libidn2```

(libidn is not installed) and got : 
```
==> Upgrading 1 outdated package:
libidn2 2.3.4 -> 2.3.4_1
==> Downloading https://ghcr.io/v2/homebrew/core/libidn2/manifests/2.3.4_1-1
Already downloaded: /Users/graham/Library/Caches/Homebrew/downloads/03ad193177f4e7d05ee2ed19a455028cb5fbf7ea1a812d88f18f5e9e8b4a4d43--libidn2-2.3.4_1-1.bottle_manifest.json
==> Fetching libidn2
==> Downloading https://ghcr.io/v2/homebrew/core/libidn2/blobs/sha256:69b39ffdb1367eabb34c77ce8bac7640d9fddc1722c95d2f2ce9ff9c4cf44db6
################################################################################################################################################################################## 100.0%
==> Upgrading libidn2
  2.3.4 -> 2.3.4_1 
```
and now wget is working. Thanks. So libidn2 is the culprit. It should have been updated as a dependency when I ran
```brew install wget```
right at the start of this, and got
```Running `brew update --auto-update`...```
which installed or upgraded a whole pile of libraries, but not libidn2.

**Comment by amadzarak** [comment]
Confirming this solution works for me.
`brew install libidn2`

**Comment by tonydeck0506** [comment]
brew install wget works for me

**Reply by cho-m** [reply]
This issue was fixed in https://github.com/Homebrew/brew/pull/16152
